Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -87,10 +86,14 @@ public AbstractCompactRandomCutTree(int maxSize, long seed, LeafStore leafStore,
         this.rootIndex = rootIndex;
         // random = new Random(seed);
         nodeManager = new CompactNodeManager(nodeStore, leafStore, maxSize);
-        nodeView = new CompactNodeView(this, rootIndex);
         this.enableCache = enableCache;
     }
 
+    @Override
+    INode<Integer> getNodeView(Integer node) {
```

Comment:
Shouldn't `getNodeView` return an `INodeView` and not an `INode`?

Reply:
The function should be a getNode(node) because the priviledges correspond to INode and more than just view.